### PR TITLE
PUBDEV-7190 - Add H2O import folder support to GCS

### PR DESCRIPTION
**Pull request by Pscheidl**
https://0xdata.atlassian.net/browse/PUBDEV-7190

### CODE REVIEW

File: h2o-app/build.gradle [hunk 1]
```diff
@@ -13,6 +13,7 @@ dependencies {
   if (project.hasProperty("doIncludeMojoPipeline") && project.doIncludeMojoPipeline == "true") {
     compile project(":h2o-ext-mojo-pipeline")
   }
+  compile project(":h2o-persist-gcs")
```
Comment (Pscheidl) [inline h2o-app/build.gradle]: Intentional. Let's keep it there for easier development, may I ?

File: h2o-persist-gcs/build.gradle [hunk 2]
```diff
@@ -4,7 +4,7 @@ description = "H2O Persist GCS"
 
 dependencies {
     compile project(":h2o-core")
-    compile 'com.google.cloud:google-cloud-storage:1.19.0'
+    compile 'com.google.cloud:google-cloud-storage:1.103.0'
```
Comment (Pscheidl) [inline h2o-persist-gcs/build.gradle]: Upgraded version, let's provide the latest fixes.

File: h2o-persist-gcs/src/main/java/water/persist/PersistGcs.java [hunk 3]
```diff
@@ -157,11 +158,18 @@ public void importFiles(String path,
...
     } else {
       try {
-        Blob blob = storageProvider.getStorage().get(bk[0], bk[1]);
-        final GcsBlob gcsBlob = GcsBlob.of(blob.getBlobId());
-        final Key k = GcsFileVec.make(path, blob.getSize());
-        keys.add(k.toString());
-        files.add(path);
+        Iterable<Blob> values = storageProvider.getStorage().list(bk[0], Storage.BlobListOption.prefix(bk[1])).getValues();
+        values.forEach(blob -> {
+          final String blobPath = new StringBuilder("gs://")
```
Comment (honzasterba) [inline h2o-persist-gcs/src/main/java/water/persist/PersistGcs.java]: if you would do just plain string concat here javac will translate it to this exact code

File: h2o-app/build.gradle (hunk 1 above)
Comment (honzasterba) [reply h2o-app/build.gradle]: what about the other persists, should they be here as well? is there a reason they are not and are added only to assembly?

Comment (Pscheidl) [reply h2o-app/build.gradle]: I have no idea about the reason. @michalkurka please ?

File: h2o-persist-gcs/src/main/java/water/persist/PersistGcs.java (hunk 3 above)
Comment (Pscheidl) [reply h2o-persist-gcs/src/main/java/water/persist/PersistGcs.java]: Ok, cool. didn't know javac will output the same bytecode. Heard of it.

Done. Thank you.

File: h2o-app/build.gradle (hunk 1 above)
Comment (Pscheidl) [reply h2o-app/build.gradle]: For now, I removed it from the dependencies.

File: h2o-hadoop-common/tests/python/pyunit_gcs_import.py [hunk 4]
```diff
@@ -1,18 +0,0 @@
-#! /usr/env/python
-
-import sys
-import os
-sys.path.insert(1, os.path.join("../../../h2o-py"))
-from tests import pyunit_utils
-import h2o
-
-def gcs_import():
-    # Just test the import works - no class clashes, no exception
```
Comment (honzasterba) [inline h2o-hadoop-common/tests/python/pyunit_gcs_import.py]: I think we should keep running this test on hdp, why not?

Comment (Pscheidl) [reply h2o-hadoop-common/tests/python/pyunit_gcs_import.py]: Oh my god, I did not intend to move it.

Comment (Pscheidl) [reply h2o-hadoop-common/tests/python/pyunit_gcs_import.py]: Moved back.

**Pull request merged**